### PR TITLE
Modules and roles changes

### PR DESCRIPTION
**Pull request by bilaluplogic**
Following changes
- Added table_schema-def.xml for modules
- Modules xarinit.php changes for new installation 
- Roles xarinit.php file changed
- LONGTEXT added in xml2ddl-mysql.xsl

### CODE REVIEW

File: html/code/modules/modules/xarinit.php [hunk 1]
```diff
@@ -20,47 +20,41 @@
...
+        // We're done, commit
+        $dbconn->commit();
+    } catch (Exception $e) {
+        $dbconn->rollback();
+        throw $e;
+    }
+    // Get database information
+   // $dbconn = xarDB::getConn();
+    $tables =& xarDB::getTables();
     $tables['eventsystem'] = $prefix . '_eventsystem';
```
Comment (xaraya) [inline html/code/modules/modules/xarinit.php]: Do we need this line?

File: html/code/modules/modules/xarinit.php [hunk 2]
```diff
@@ -20,47 +20,41 @@
...
-         *   class varchar(64) NOT NULL,
-         *   category varchar(64) NOT NULL,
-         *   admin_capable INTEGER NOT NULL default '0',
-         *   user_capable INTEGER NOT NULL default '0',
-         *   state INTEGER NOT NULL default '0'
-         *   PRIMARY KEY  (id)
-         * )
-         */
-        $fields = array(
+       /* $fields = array(
```
Comment (xaraya) [inline html/code/modules/modules/xarinit.php]: No need to comment out. Just remove the unneeded code

File: html/code/modules/modules/xarinit.php [hunk 3]
```diff
@@ -218,14 +162,15 @@ class, category, admin_capable, user_capable, state )
         }
         // We're done, thanks, commit the thingie
         $dbconn->commit();
-    } catch (Exception $e) {
+    /*} catch (Exception $e) {
```
Comment (xaraya) [inline html/code/modules/modules/xarinit.php]: Why are we removing this?

File: html/code/modules/modules/xarinit.php [hunk 4]
```diff
@@ -235,6 +180,8 @@ class, category, admin_capable, user_capable, state )
  */
 function modules_activate()
 {
+
```
Comment (xaraya) [inline html/code/modules/modules/xarinit.php]: Why the extra lines?

File: html/code/modules/roles/xardata/table_schema-def.xml [hunk 5]
```diff
@@ -6,31 +6,31 @@
       <number size="10" unsigned="true" />
     </column>
     <column id="roles.name" name="name" required="true" >
-      <text size="255" charset="utf8"/>
+      <text size="255" charset=" utf8"/>
```
Comment (xaraya) [inline html/code/modules/roles/xardata/table_schema-def.xml]: Remove these spaces. I have adjusted the XSL to add a space.

File: html/code/modules/roles/xarinit.php [hunk 6]
```diff
@@ -230,23 +231,24 @@ function roles_activate()
     $rolefields['name'] = 'Anonymous';
     $rolefields['uname'] = 'anonymous';
     $rolefields['parentid'] = $topid;
-    $anonid = $user->createItem($rolefields);
+    $anonid = $group->createItem($rolefields);
```
Comment (xaraya) [inline html/code/modules/roles/xarinit.php]: This is a user, not a group!

File: html/code/modules/roles/xarinit.php [hunk 7]
```diff
@@ -230,23 +231,24 @@ function roles_activate()
...
+    $adminid = $group->createItem($rolefields);
     xarModVars::set('roles', 'admin', $adminid);
 
     // The SiteManager
     $rolefields['name'] = 'SiteManager';
     $rolefields['uname'] = 'manager';
     $rolefields['email'] = 'none@none.com';
     $rolefields['parentid'] = $mgrgroup;
-    $mgrid = $user->createItem($rolefields);
+    $mgrid = $group->createItem($rolefields);
```
Comment (xaraya) [inline html/code/modules/roles/xarinit.php]: This is a user, not a group!

**Review by xaraya — CHANGES_REQUESTED** [review]
Please make the changes indicated in my comments

File: html/code/modules/modules/xarinit.php (hunk 1 above)
Comment (bilaluplogic) [reply html/code/modules/modules/xarinit.php]: We need this to overcome following error:
File     : html/lib/xaraya/events.php
Line     : 509
Undefined index: eventsystem

File: html/code/modules/modules/xarinit.php (hunk 2 above)
Comment (bilaluplogic) [reply html/code/modules/modules/xarinit.php]: okay we will remove.

File: html/code/modules/modules/xarinit.php (hunk 3 above)
Comment (bilaluplogic) [reply html/code/modules/modules/xarinit.php]: okay I will change it.

File: html/code/modules/modules/xarinit.php (hunk 4 above)
Comment (bilaluplogic) [reply html/code/modules/modules/xarinit.php]: okay i will remove.

File: html/code/modules/roles/xardata/table_schema-def.xml (hunk 5 above)
Comment (bilaluplogic) [reply html/code/modules/roles/xardata/table_schema-def.xml]: okay I will do it.

File: html/code/modules/roles/xarinit.php (hunk 6 above)
Comment (bilaluplogic) [reply html/code/modules/roles/xarinit.php]: okay i will change it.

File: html/code/modules/roles/xarinit.php (hunk 7 above)
Comment (bilaluplogic) [reply html/code/modules/roles/xarinit.php]: okay

**Pull request merged**